presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -378,6 +378,7 @@ public PlanOptimizers(
                         statsCalculator,
```

Comment:
What is the benefit or motivation behind this change?

Reply:
The rule `TransformCorrelatedScalarAggregationToJoin` matches to a CorrelatedJoin, which has a scalar AggregationNode in the subquery. It matches also in the case, when there are multiple Projections in the subquery above the AggregationNode.

As the rule is applied, the Aggregation is pulled up above Join, and the Projection is restored on top of the aggregation. The issue is that in the case of multiple Projections, only the top one is regarded. It might cause a failure due to missing dependencies.

The proposed solution is not to apply the rule if there are multiple Projections, but let `InlineProjections` squash them first.